### ISSUE TITLE
Add Appimage package for portability on Linux

### ISSUE THREAD

**Issue by reyespinosa1996** [ezra-bible-app/ezra-bible-app#140]
Should be great to have ezra release as an appimage. Because users don't using Fedora, Ubuntu, Debian or Arch could still use Ezra.

**Comment by tobias-klein** [comment]
@reyespinosa1996 Good idea! Out of curiosity - which distribution are you using? Could you imagine to provide this feature yourself and submit a pull request?

**Comment by tobias-klein** [comment]
I tried this with electron-builder, but there seems to an issue to copy all necessary shared libraries into the AppImage. So now either electron-builder must be fixed, or this could be manually just with the AppImage tools somehow.

**Comment by tobias-klein** [comment]
This turns out to be more complicated than you may think.
Also the manual AppImage building approach failed (see the [AppImage2 branch](https://github.com/ezra-project/ezra-project/tree/AppImage2)). The resulting AppImage did not run on Linux versions with older packages, like Debian 10 and CentOS 8. With that said I'm marking this issue as wontfix. I personally won't invest more time here for now. If somebody comes up with a solution, however - pull requests are welcome!

**Comment by alerque** [comment]
AppImage is often peddled as some kind of panacea solving distribution comparability issues. My experience is that it's more of a chimaera with a pretty face in a few cases that hides a whole rash of edge cases it actually exacerbates rather than alleviates. Of course anyone is welcome to contribute whatever piece they really want to see, but I would suggest Tobias is right not to invest more time in this direction. It's easier to support directly as a package than as a package container.

**Comment by reyespinosa1996** [comment]
> @reyespinosa1996 Good idea! Out of curiosity - which distribution are you using? Could you imagine to provide this feature yourself and submit a pull request?

I'm currently using OpenSuse Leap.

**Comment by alerque** [comment]
It shouldn't be *too* hard to setup a native OpenSUSE package. I've done it before but it's been quite a while and I think their infrastructure has changed significantly since my days. We'd want a similar application (Electron app that uses electron-builder) to compare packaging notes with. Have you ever done packaging before? Or is there anybody else out there following this issue with OpenSUSE experience?

**Comment by reyespinosa1996** [comment]
I've never did packaging task before. But it is something I'd like trying to do.  Maybe I will and send a PR here with the new package.

**Comment by tobias-klein** [comment]
Actually, there has been a working OpenSuse Leap package before (until version [0.13.2](https://github.com/ezra-project/ezra-project/releases/tag/0.13.2)). I think I stopped shipping that because there was some issue in generating the package most recently, I do not remember why at this point. I'm going to create a branch to re-enable the package creation and see whether the issue can be resolved. Note that one special thing about OpenSuse is that packages should ideally be signed. That's certainly not the case for the Ezra Project package I generated and users will get a warning when they attempt to install. But I know that the package worked.
